### PR TITLE
rho grid bug fixes

### LINKED ISSUE #619: BootstrapRedlConsistency default grid
The default grid for the `BootstrapRedlConsistency` objective includes the $\rho=1$ surface. Typical density and pressure profiles vanish at the boundary surface; this causes division by 0 in the Redl formula, and results in the objective returning a `NaN` value. 

The easy fix is to change the default grid to exclude the $\rho=1$ surface. We may also want to do some checking when the objective is built to ensure the profiles and grid are consistent with each other.

### PR DESCRIPTION
**Pull request by ddudt**
Resolves #619 and also fixes a bug where `PowerSeriesProfile.from_values(sym=True)` did not actually return an even profile.

### CODE REVIEW

**Review by dpanici — CHANGES_REQUESTED** [review]
Can you add a test for the PowerSeriesProfile fix? like just going from SplineProfile to PowerSeries and checking that it is even I mean

File: desc/profiles.py [hunk 1]
```diff
@@ -768,6 +766,9 @@ def from_values(
             profile in power series basis fit to given data.
 
         """
+        if sym:
+            x = x**2
+            order = order // 2
```
Comment (f0uriest) [inline desc/profiles.py:771]: I don't think we want to reduce the order here without padding the params later, or passing in the mode numbers explicitly

**Review by f0uriest — CHANGES_REQUESTED** [review]
I think the even profile fix isn't quite right. Also we should leave #619 open for now, or make a new issue to evaluate the limit properly

File: tests/test_profiles.py [hunk 2]
```diff
@@ -346,16 +360,6 @@ def test_profile_errors(self):
             a = sp * pp
             a.params = sp.params
 
-    @pytest.mark.unit
```
Comment (ddudt) [inline tests/test_profiles.py:349]: I deleted this test because it was redundant with `test_close_values`.

**Review by f0uriest — APPROVED** [review]

**Review by dpanici — APPROVED** [review]

**Pull request merged**